drivers/amazonec2/amazonec2.go

```diff
@@ -1109,37 +1114,35 @@ func (d *Driver) securityGroupAvailableFunc(id string) func() bool {
 	}
 }
 
-func (d *Driver) configureTags(tagGroups string) error {
-
-	tags := []*ec2.Tag{}
-	tags = append(tags, &ec2.Tag{
-		Key:   aws.String("Name"),
-		Value: &d.MachineName,
-	})
-
-	if tagGroups != "" {
-		t := strings.Split(tagGroups, ",")
-		if len(t) > 0 && len(t)%2 != 0 {
-			log.Warnf("Tags are not key value in pairs. %d elements found", len(t))
-		}
-		for i := 0; i < len(t)-1; i += 2 {
-			tags = append(tags, &ec2.Tag{
-				Key:   &t[i],
-				Value: &t[i+1],
+// buildResourceTags accepts a list of AWS resources that should be tagged
+// upon creation of the EC2 instance. Driver.Tags will be applied to all resources
+// supplied, except for the ec2InstanceResource which will also have the
+// MachineName added as a tag.
+//
+// NB: The ec2InstanceResource must be passed for the EC2 instance to have a name.
+func (d *Driver) buildResourceTags(resources []string) []*ec2.TagSpecification {
+	tags := buildEC2Tags(d.Tags)
+	var tagSpecs []*ec2.TagSpecification
+	for i := 0; i < len(resources); i++ {
+		switch resources[i] {
+		case ec2InstanceResource:
+			// append instance name
+			instanceTags := append(tags, &ec2.Tag{
+				Key:   aws.String("Name"),
+				Value: &d.MachineName,
+			})
+			tagSpecs = append(tagSpecs, &ec2.TagSpecification{
+				ResourceType: &resources[i],
+				Tags:         instanceTags,
+			})
+		default:
+			tagSpecs = append(tagSpecs, &ec2.TagSpecification{
+				ResourceType: &resources[i],
+				Tags:         tags,
```

Comment:
This may be a little simpler.
```suggestion
		if resources[i] == ec2InstanceResource:
			// append instance name
			instanceTags := append(tags, &ec2.Tag{
				Key:   aws.String("Name"),
				Value: &d.MachineName,
			})
		}
		tagSpecs = append(tagSpecs, &ec2.TagSpecification{
			ResourceType: &resources[i],
			Tags:         instanceTags,
		})
```